source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -149,6 +149,9 @@ RedisCluster::DnsDiscoveryResolveTarget::~DnsDiscoveryResolveTarget() {
   if (active_query_) {
     active_query_->cancel();
   }
+  if (resolve_timer_) {
+    resolve_timer_->disableTimer();
+  }
```

Comment:
All of these are done implicity by the destructor. Did you do these for testing? If so make sure you have expectations and add a comment above each disable saying you are doing this for ease of testing. Same elsewhere.

Reply:
Ping on this one.

Reply:
I'm not sure I understand how this is done by the destructor.  The disableTimer() calls event_del(&raw_event_), but the destructor doesn't.  https://github.com/envoyproxy/envoy/blob/a78311faf214bb9216f92407102b547b6fcd14a3/source/common/event/timer_impl.cc#L30
I also see the same pattern in a few other places.
The resolve_timer_ disableTimer() expectation is already created in the test.  https://github.com/envoyproxy/envoy/blob/9c0681d8be803008cfdc1366aa2cd20cd0354283/test/extensions/clusters/redis/redis_cluster_test.cc#L163

Reply:
The destructor disables the event. The reason that disable is explicitly called is so that it can be mocked. So please add a comment and/or remove if you are not checking it in a mock.

/wait